### ISSUE TITLE
Await refresh issue ?

### ISSUE THREAD

**Issue by sparqueur** [FluentLenium/FluentLenium#478]
Hi,

I used to wait for dom conditions using the following :
await().atMost(20, TimeUnit.SECONDS).until($("#toto")).present()
It seems that if the condition is not fullfield at first evalution, result will always be bad as if it was not evaluate again.
For instance, if put a TimeUnit.SECONDS.sleep(2) just before, my test is in success. (remark : I obviously don't need 22 seconds)
I am doing something wrong ?

My current configuration :
- Chrome with latests driver
- Selenium : 3.0.1
- Fluentlenium : 3.1.1
Same issue with firefox

I previously used fluentlenium 1.0.RC1 with the same issue but it's been really really worse seens I upgraded to 3.0.1.

Thanks in advance

**Comment by filipcynarski** [comment]
@sparqueur thank you for rising the issue. It may expose the issue because FluentLenium is much faster now. Could you please provide URL to tested site? It may because you need to wait for something and test may require some adjustments. Basing on provided data I'm not able to proceed w/ any further steps or investigation.

In my project (~600 FluentLenium scenarios) we had to improve our tests because waits was implemented improperly.

If you run your test in debug mode it will pass, so my assumption is you need to work on your await object, but I can dig deeper If you provide your testing object and code to evaluate (you can extract to to example project).

**Comment by sparqueur** [comment]
Sorry, I won't be able to provide an example.
I currently use a workaround : using awaitility : 
Awaitility.waitAtMost(duration, timeUnit).until(() -> $("#toto")).present() )
It seems to work.
But can you confirm me that there is some kind of "cache" when calling ".present()"
It does not always look at the dom but look for a known proxy.
==> If you manually touch the gui while running, you can break the cache
For instance, if I manually close an element (remove from dom by doing an action) and I call ".present()" it may return true.
Am I wrong ?
What about "dynamic dom" for instance on rest callback ?

**Comment by filipcynarski** [comment]
Have you seen documentation section about lazy locators?
We are using proxy to elements, if the list was filled in and you need to check if size changed for example you need to reset the proxy reference or call it using lazy locators.

There is docs section called: `Lazy Fluent Locators`

```
await().atMost(20, TimeUnit.SECONDS).until($("#toto")).present()
```

I'm using such constructions many times and it is working for me. Have you tried displayed()

instead of using Awaitility library you can write your own predicate.

```
await().untilPredicate((fl) ->$("#toto").present());
```

**Comment by filipcynarski** [comment]
The issue will be fixed as a part of #489
